internal/registry/registry_helper_test.go

```diff
@@ -0,0 +1,66 @@
+package registry
```

Comment:
Why don't we put that code as part of `internal/registry/registry_test.go`?

Reply:
since it is quite a few implementation details for mocking the layering, it was better to move it to a helper file, and leave it separate from the test code itself